src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -543,6 +533,43 @@ async Task<ElementIndexInfo> GetElementIndexInfo(List<JObject> nestedIndexers)
                     ElementIdxStr: elementIdxStr.ToString(),
                     Indexers: indexers);
             }
+        }
+
+        private async Task<JObject> InvokeGetItemOnJObject(
+            JObject rootObject,
+            int typeId,
+            DotnetObjectId objectId,
+            ElementIndexInfo elementIdxInfo,
+            CancellationToken token)
+        {
+            int[] methodIds = await context.SdbAgent.GetMethodIdsByName(typeId, "get_Item", BindingFlags.Default, token);
+            if (methodIds == null || methodIds.Length == 0)
+                throw new InvalidOperationException($"Type '{rootObject?["className"]?.Value<string>()}' cannot be indexed.");
+            var type = rootObject?["type"]?.Value<string>();
+
+            // ToDo: optimize the loop by choosing the right method at once without trying out them all
```

Comment:
What is needed, or missing to do this?

Reply:
Your idea here is the better approach.

Reply:
Reason why it might be not possible:
1) If the parameter is required (as opposed to default) then we don't have information about its type. Even though we know what types were passed to the method, we cannot tell if this method will work well with them till we ask the runtime to `CMD_VM_INVOKE_METHOD` and it will throw / return a result.
2) Sometimes we even are not sure what type was passed to the method, see https://github.com/dotnet/runtime/issues/93057.
It would be good to try if something changed but please, not in this PR. I will see if I can make it in a separate one.